src/move/coin_factory/sources/coin_factory.move

```diff
@@ -0,0 +1,92 @@
+module coin_factory::coin_factory {
+    use aptos_std::coin::{Self, BurnCapability, FreezeCapability, MintCapability};
+    use aptos_std::string::{Self, String};
+    use std::signer;
+    use std::vector;
+    use lp_coin_manager::lp_coin_manager;
+
+    struct Emojicoin {}
+    struct EmojicoinLP {}
+
+    const EMOJICOIN_SUPPLY: u64 = 450_000_000_000_000_000;
```

Comment:
Adapt https://github.com/econia-labs/emojicoin-dot-fun/pull/10#discussion_r1571107829 for this case